src/Layout/redist/targets/GenerateMSIs.targets

```diff
@@ -233,12 +242,45 @@
+  <Target Name="UnzipBuiltinWorkloadManifestMsiNuPkg">
+    <PropertyGroup>
+      <BuiltinManifestMsiPackageSourceDirectory>$(ArtifactsShippingPackagesDir)</BuiltinManifestMsiPackageSourceDirectory>
+      <BuiltinManifestMsiPackageSourceDirectory Condition="'$(DotNetBuild)' != 'true' and '$(BuildWorkloads)' != 'true' and '$(OfficialBuildId)' != ''">$(ArtifactsDir)downloaded-manifest-msi-packages\PackageArtifacts</BuiltinManifestMsiPackageSourceDirectory>
+      <BuiltinManifestMsiPackageSourceDirectory Condition="'$(DotNetBuild)' == 'true' and '$(DotNetBuildPass)' == '3'">$(SourceBuiltShippingPackagesDir)</BuiltinManifestMsiPackageSourceDirectory>
+    </PropertyGroup>
+
+    <ItemGroup>
+      <_builtinManifestMsiPackages Include="$(BuiltinManifestMsiPackageSourceDirectory)/%(BuiltinManifests.Identity).Manifest-*.Msi.$(MsiArchitectureForWorkloadManifests)*.nupkg" />
+      <BuiltinManifestMsiPackages Include="@(_builtinManifestMsiPackages)" UnzipDestinationFolder="$(IntermediateOutputPath)\extracted-manifests\%(Filename)" />
+    </ItemGroup>
+
+    <PropertyGroup>
+      <!-- for local SDK builds we allow skipping the workload manifest msi generation to save build time, the sdk bundle won't contain the manifest msis in that case. -->
+      <AllowEmptyBuiltinWorkloadManifestMsi Condition="'$(DotNetBuild)' != 'true' and '$(BuildWorkloads)' != 'true' and '$(OfficialBuildId)' == ''">true</AllowEmptyBuiltinWorkloadManifestMsi>
+    </PropertyGroup>
+
+    <Warning Text="Builtin manifest msi packages were skipped, the sdk bundle won't contain the builtin manifests and workload installs will be broken."
+             Code="MISSING_MANIFEST_MSI"
+             Condition="@(BuiltinManifestMsiPackages -> Count()) == 0 and '$(AllowEmptyBuiltinWorkloadManifestMsi)' == 'true'" />
+
+    <Error Text="Could not find expected builtin manifest msi packages in $(BuiltinManifestMsiPackageSourceDirectory)"
+           Condition="@(BuiltinManifestMsiPackages -> Count()) == 0 and '$(AllowEmptyBuiltinWorkloadManifestMsi)' != 'true'" />
+
+    <Unzip SourceFiles="@(BuiltinManifestMsiPackages)"
+           DestinationFolder="%(BuiltinManifestMsiPackages.UnzipDestinationFolder)"
+           Condition="@(BuiltinManifestMsiPackages -> Count()) != 0" />
+  </Target>
+
+  <Target Name="GenerateWorkloadManifestsWxs" DependsOnTargets="UnzipBuiltinWorkloadManifestMsiNuPkg">
     <ItemGroup>
       <_Manifests Include="@(BundledManifests)">
-        <RestoredMsiPathInNupkg>$([MSBuild]::NormalizePath($([System.IO.Directory]::GetFiles('%(RestoredMsiNupkgContentPath)/data/', '*$(MsiArchitectureForWorkloadManifests).msi'))))</RestoredMsiPathInNupkg>
+        <MsiPath>$([MSBuild]::NormalizePath($([System.IO.Directory]::GetFiles('%(RestoredMsiNupkgContentPath)/data/', '*$(MsiArchitectureForWorkloadManifests).msi'))))</MsiPath>
+      </_Manifests>
+      <_Manifests Include="@(BuiltinManifestMsiPackages)" Condition="@(BuiltinManifestMsiPackages -> Count()) != 0">
```

Comment:
This condition is a bit unusual. It shouldn't be necessary to condition on the Count. If an item is empty, the Include will just be a no-op.

Reply:
I tried that but it still tries to evaluate the `MsiPath` attribute in that case, resulting in
> error MSB4184: The expression "[System.IO.Directory]::GetFiles(/data/, *.msi)" cannot be evaluated. Could not find a part of the path '/data'.

Reply:
Oh, so when the item set is empty, msbuild then treats this as an update. Probably an msbuild bug.